ProcessLib/HeatTransportBHE/BHE/CreateFlowAndTemperatureControl.cpp

```diff
@@ -0,0 +1,90 @@
+/**
+ * \file
+ *
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "BaseLib/ConfigTree.h"
+#include "BaseLib/Error.h"
+
+#include "CreateFlowAndTemperatureControl.h"
+#include "RefrigerantProperties.h"
+
+namespace ProcessLib
+{
+namespace HeatTransportBHE
+{
+namespace BHE
+{
+FlowAndTemperatureControl createFlowAndTemperatureControl(
+    BaseLib::ConfigTree const& config,
+    std::map<std::string,
+             std::unique_ptr<MathLib::PiecewiseLinearInterpolation>> const&
+        curves,
+    RefrigerantProperties const& refrigerant)
+{
+    auto find_curve_or_error = [&](std::string const& name,
```

Comment:
You can use the very similar function `getOrError()` from `BaseLib/Algorithm.h`.

Reply:
Thanks for your comment. Using the existed function getOrError().